File: 백준/Silver/1002. 터렛/터렛.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        while (n-- > 0) {
            int x1 = sc.nextInt();
            int y1 = sc.nextInt();
            int r1 = sc.nextInt();
            int x2 = sc.nextInt();
            int y2 = sc.nextInt();
            int r2 = sc.nextInt();
            double d = Math.sqrt(Math.pow(x1 - x2, 2) + Math.pow(y1 - y2, 2));
            int bigR, smallR;
            if (r1 > r2) {
                bigR = r1;
                smallR = r2;
            } else {
                bigR = r2;
                smallR = r1;
            }
            if (d == 0 && r1 == r2)
                System.out.println("-1");
            else if (d > r1 + r2 || bigR > smallR + d)
                System.out.println("0");
            else if (d == r1 + r2 || bigR == smallR + d)
                System.out.println("1");
            else
                System.out.println("2");
        }
    }
}

File: 백준/Silver/1002. 터렛/터렛.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);
    cout.tie(0);
    
    int n, x1, y1, r1, x2, y2, r2;
    double d;
    cin >> n;
    while (n--) {
        cin >> x1 >> y1 >> r1 >> x2 >> y2 >> r2;
        d = sqrt(pow(x1 - x2, 2) + pow(y1 - y2, 2));
        int bigR, smallR;
        bigR = r1 > r2 ? r1 : r2;
        smallR = r1 < r2 ? r1 : r2;
        if (d == 0 && bigR == smallR)
            cout << "-1\n";
        else if (d > bigR + smallR || bigR > smallR + d)
            cout << "0\n";
        else if (d == bigR + smallR || bigR == smallR + d)
            cout << "1\n";
        else
            cout << "2\n";
    }
}